sunpy/net/helioviewer.py

```diff
@@ -450,3 +435,15 @@ def _request(self, params):
     def _format_date(self, date):
         """Formats a date for Helioviewer API requests"""
         return parse_time(date).isot + "Z"
+
+    def _get_source_id(self, key):
+        """
+        Returns source_id based on the key.
+        """
+        source_id_list = list(partial_match(key,self.data_sources))
+        if len(source_id_list) == 1:
+            return source_id_list[0]
+        else:
+            raise KeyError("The values used: {observatory}, {instrument}, {detector}, {measurement} "
+                            "do not correspond to one source_id but str(len(source_id_list)) source_id(s)."
+                            " Please check the list using HelioviewerClient.data_sources.")
```

Comment:
How does this error message look like now?

Reply:
I will upload a screen shot

Reply:
![error](https://user-images.githubusercontent.com/31286078/52740269-aea5b100-2ff8-11e9-9e32-a797834bf109.png)

Reply:
You need a `f`  in front of the string. It has not rendered correctly.

Reply:
I did it with `f` but there was no change

Reply:
You did `(f"`?

Reply:
If that doesn't work, we will need to use `.format()` instead.

Reply:
Oh ok I will go with format because the `f` didn't work for me

Reply:
What exactly do you want it to render? Do we need the values of `observatory`, `instrument` etc to be shown in the error message?

Reply:
![error](https://user-images.githubusercontent.com/31286078/52742656-3641ee80-2ffe-11e9-8689-e4e42c82cfbd.png)
Is this ok?

Reply:
Yeah that looks like it works now.

Reply:
it was not recognising observatory etc but when I replaced it with {key} it worked